dev/tasks/docker-tests/github.cuda.yml

```
@@ -0,0 +1,48 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+{% import 'macros.jinja' as macros with context %}
+
+{{ macros.github_header() }}
+
+jobs:
+  test:
+    name: |
+      Docker Test {{ flags|default("") }} {{ image }} {{ command|default("") }}
+    runs-on: ['self-hosted', 'cuda'] 
+{{ macros.github_set_env(env) }}
+    steps:
+      {{ macros.github_checkout_arrow(fetch_depth=fetch_depth|default(1))|indent }}
+      # python 3.8 is installed on the runner, no need to install
+      - name: Install archery
+        run: python -m pip install -e arrow/dev/archery[docker]
+      - name: Execute Docker Build
+        shell: bash
+        env:
+        {{ macros.github_set_sccache_envvars()|indent(8) }}
+        run: |
+          archery docker run \
```

So that we can see what the executed docker commands look like
```suggestion
          archery docker --debug run \
```

---

I'll add it manualy to the workflow so we don't have to wait

---

https://github.com/ursacomputing/crossbow/actions/runs/3321733516/jobs/5489783547

---

https://github.com/ursacomputing/crossbow/actions/runs/3321733516/jobs/5489783547#step:5:18

---

It looks like it is running it via docker cli instead of docker-compose: 
```
DEBUG:archery:Executing `['docker', 'run', '--rm', '-e', 'SETUPTOOLS_SCM_PRETEND_VERSION=10.0.0.dev509', '--gpus', 'all', '--shm-size', '2G', '-e', 'ARROW_BUILD_UTILITIES=OFF', '-e', 'ARROW_COMPUTE=OFF', '-e', 'ARROW_CSV=OFF', '-e', 'ARROW_CUDA=ON', '-e', 'ARROW_DATASET=OFF', '-e', 'ARROW_ENABLE_TIMING_TESTS=OFF', '-e', 'ARROW_FILESYSTEM=OFF', '-e', 'ARROW_GANDIVA=OFF', '-e', 'ARROW_GCS=OFF', '-e', 'ARROW_HDFS=OFF', '-e', 'ARROW_JEMALLOC=OFF', '-e', 'ARROW_JSON=OFF', '-e', 'ARROW_ORC=OFF', '-e', 'ARROW_PARQUET=OFF', '-e', 'ARROW_PLASMA=OFF', '-e', 'ARROW_S3=OFF', '-e', 'ARROW_SUBSTRAIT=OFF', '-e', 'ARROW_WITH_OPENTELEMETRY=OFF', '-e', 'AWS_ACCESS_KEY_ID=None', '-e', 'AWS_SECRET_ACCESS_KEY=None', '-e', 'CCACHE_COMPILERCHECK=content', '-e', 'CCACHE_COMPRESS=1', '-e', 'CCACHE_COMPRESSLEVEL=6', '-e', 'CCACHE_DIR=/ccache', '-e', 'CCACHE_MAXSIZE=1G', '-e', 'SCCACHE_BUCKET=None', '-e', 'SCCACHE_S3_KEY_PREFIX=sccache', '-v', '/runner/_work/crossbow/crossbow/arrow:/arrow:delegated', '-v', 'ubuntu-ccache:/ccache:delegated', 'apache/arrow-dev:amd64-ubuntu-20.04-cuda-11.0.3-cpp', '/bin/bash', '-c', '\n  /arrow/ci/scripts/cpp_build.sh /arrow /build &&\n  /arrow/ci/scripts/integration_skyhook.sh /build &&\n  /arrow/ci/scripts/cpp_test.sh /arrow /build']`
```

---

same error when running with docker-compose directly https://github.com/ursacomputing/crossbow/actions/runs/3321875304/jobs/5490121964#step:5:2730

---

Can you reproduce directly by ssh'ing into the runner?